02_bubble/lokili/main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(arr []int) []int {
+	arrLen := len(arr)
+	sorted := make([]int, arrLen)
+	copy(sorted, arr)
+
+	for i := 0; i < arrLen-1; i++ {
+		swapped := false
+		for j := 0; j < arrLen-i-1; j++ {
+			if sorted[j] > sorted[j+1] {
+				sorted[j], sorted[j+1] = sorted[j+1], sorted[j]
+				swapped = true
+			}
+		}
+		if !swapped {
+			break
+		}
+	}
+
+	return sorted
+}
+
+func insertion(arr []int) []int {
+	arrLen := len(arr)
+	sorted := make([]int, arrLen)
+	copy(sorted, arr)
+
+	for i := 1; i <= arrLen-1; i++ {
+		insertValue := sorted[i]
+		j := i - 1
+		for j >= 0 && sorted[j] > insertValue {
+			sorted[j+1] = sorted[j]
+			j--
+		}
+		sorted[j+1] = insertValue
+	}
+	return sorted
+}
+
+func mergeSort(arr []int, start int, mid int, end int, tmp []int) {
```

Comment:
```go
…, start, mid, end int, …
```

Reply:
Sorry for forgetting the go format. Change it.